src/hotspot/share/opto/superword.cpp

```diff
@@ -2237,396 +1992,65 @@ void SuperWord::apply_memops_reordering_with_schedule(Node_List& memops_schedule
-      if (VerifyAlignVector) {
-        if (vn->Opcode() == Op_LoadVector) {
-          vn->as_LoadVector()->set_must_verify_alignment();
-        } else if (vn->Opcode() == Op_StoreVector) {
-          vn->as_StoreVector()->set_must_verify_alignment();
-        }
-      }
-#endif
+  for (int i = 0; i < _schedule.length(); i++) {
+    VTransformNode* vtn = _schedule.at(i);
+    VTransformApplyStatus status = vtn->apply(_vloop_analyzer,
+                                              vnode_idx_to_transformed_node);
+    NOT_PRODUCT( if (_is_trace_verbose) { status.trace(vtn); } )
 
-      phase()->register_new_node_with_ctrl_of(vn, first);
-      for (uint j = 0; j < p->size(); j++) {
-        Node* pm = p->at(j);
-        igvn().replace_node(pm, vn);
-      }
-      igvn()._worklist.push(vn);
+    vnode_idx_to_transformed_node.at_put(vtn->_idx, status.node());
+    max_vector_length = MAX2(max_vector_length, status.vector_length());
+    max_vector_width  = MAX2(max_vector_width,  status.vector_width());
+  }
 
-      if (vlen > max_vlen) {
-        max_vlen = vlen;
-      }
-      if (vlen_in_bytes > max_vlen_in_bytes) {
-        max_vlen_in_bytes = vlen_in_bytes;
-      }
-      VectorNode::trace_new_vector(vn, "SuperWord");
-    }
-  }//for (int i = 0; i < body().length(); i++)
+  assert(max_vector_length > 0 && max_vector_width > 0, "must have vectorized");
+  cl()->mark_loop_vectorized();
 
-  if (max_vlen_in_bytes > C->max_vector_size()) {
-    C->set_max_vector_size(max_vlen_in_bytes);
-  }
-  if (max_vlen_in_bytes > 0) {
-    cl->mark_loop_vectorized();
+  if (max_vector_width > C->max_vector_size()) {
+    C->set_max_vector_size(max_vector_width);
   }
 
   if (SuperWordLoopUnrollAnalysis) {
-    if (cl->has_passed_slp()) {
-      uint slp_max_unroll_factor = cl->slp_max_unroll();
-      if (slp_max_unroll_factor == max_vlen) {
+    if (cl()->has_passed_slp()) {
+      uint slp_max_unroll_factor = cl()->slp_max_unroll();
+      if (slp_max_unroll_factor == max_vector_length) {
 #ifndef PRODUCT
         if (TraceSuperWordLoopUnrollAnalysis) {
-          tty->print_cr("vector loop(unroll=%d, len=%d)\n", max_vlen, max_vlen_in_bytes*BitsPerByte);
+          tty->print_cr("vector loop(unroll=%d, len=%d)\n",
+                        max_vector_length,
+                        max_vector_width * BitsPerByte);
         }
 #endif
         // For atomic unrolled loops which are vector mapped, instigate more unrolling
-        cl->set_notpassed_slp();
+        cl()->set_notpassed_slp();
         // if vector resources are limited, do not allow additional unrolling
         if (Matcher::float_pressure_limit() > 8) {
           C->set_major_progress();
-          cl->mark_do_unroll_only();
-        }
-      }
-    }
-  }
-
-  return true;
-}
-
-//------------------------------vector_opd---------------------------
-// Create a vector operand for the nodes in pack p for operand: in(opd_idx)
-Node* SuperWord::vector_opd(Node_List* p, int opd_idx) {
```

Comment:
Note:
This method used to find the vector/scalar inputs for packs. These inputs can have many forms:
- Already vectorized: `opd->is_Vector() || opd->is_LoadVector()`
- `PopulateIndexNode` pattern (not packed)
- `same_input`, i.e. most likely a `scalar2vector` aka `Replicate` pattern. Some nodes require such inputs to be specially manipulated, so that makes it a bit more complicated.

The analogue is now `SuperWordVTransformBuilder::get_vtnode_vector_input_at_index`.